src/binding/petsc4py/src/petsc4py/PETSc/Mat.pyx

```diff
@@ -1503,7 +2254,42 @@ cdef class Mat(Object):
         CHKERR( MatMatTransposeMult(self.mat, mat.mat, reuse, rval, &result.mat) )
         return result
 
-    def transposeMatMult(self, Mat mat, Mat result=None, fill=None):
+    def transposeMatMult(
+        self,
+        Mat mat,
+        Mat result=None,
+        fill: float | None = None
+    ):
+        """Perform matrix-matrix multiplication C=AᵀB.
+
+        Neighbour-wise collective.
+
+        Parameters
+        ----------
+        mat
+            The right hand matrix B.
+        result
+            The resultant matrix C, can be ``None``.
+        fill
+            Expected fill as ratio of nnz(C)/(nnz(A) + nnz(B)), use
+            ``None`` if you do not have a good estimate. If the
+            result is a dense matrix this is irrelevant.
+
+        Returns
+        -------
+        result: Mat
+            The resultant product matrix C.
+
+        Notes
```

Comment:
add a note on the fact that  MAT_REUSE_MATRIX is used if the result is provided as an argument. Here and in all other MatMat operations